tests/providers/amazon/aws/hooks/test_redshift.py

```diff
@@ -103,3 +106,47 @@ def test_cluster_status_returns_available_cluster(self):
         hook = RedshiftHook(aws_conn_id='aws_default')
         status = hook.cluster_status('test_cluster')
         assert status == 'available'
+
+
+class TestRedshiftSQLHookConn(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.connection = Connection(login='login', password='password', host='host', port=5439, schema="dev")
+
+        self.db_hook = RedshiftSQLHook()
+        self.db_hook.get_connection = mock.Mock()
+        self.db_hook.get_connection.return_value = self.connection
+
+    def test_get_uri(self):
```

Comment:
If connection-parameter verification logic is added to the `RedshiftSQLHook._get_conn_params()` method, adding a  test validating the logic should be included here as well.

Reply:
added verification logic, and corresponding test in eac5cdacd75c0783f3418af24185f6e009a49c06